### DISCUSSION TITLE
Recognition model training loss does not converge

### DISCUSSION THREAD

**Discussion by lfxuan** [mindee/doctr#684] [Q&A]
When the pre training model is not loaded, ten thousand 4-character verification code images are synthesized to train the master model. The loss remains 3 unchanged. When the pre training model is loaded for training, it can be reduced to 0.1. How to solve the problem of loss non convergence without loading the pre training model?

**Comment by charlesmindee** [comment]
Hi @lfxuan,
Which framework are you using ?

**Comment by lfxuan** [comment]
Hello! @charlesmindee I tried every model, in which the loss of master model hovered around 3, the loss of SAR resnet31 model hovered around 4, and the loss of crnn model hovered around 17.

**Reply by charlesmindee** [reply]
Hello again @lfxuan, are you using the tensorflow backend or the torch backend ?

**Reply by lfxuan** [reply]
@charlesmindee I use tensorflow backend and vacob to reset to the required characters, and add training data for training. The following training settings and loss output:

Namespace(arch='master', batch_size=80, epochs=30, input_size=40, lr=0.001, name=None, pretrained=False, resume=None, show_samples=False, test_only=False, train_path='***/doctr/train_data/rec_data/train_set_million', val_path='***/doctr/train_data/rec_data/valid_set_million', vocab='la', wb=False, workers=None)
Validation set loaded in 2.717s (171232 samples in 2141 batches)
Train set loaded in 4.265s (696573 samples in 8707 batches)

Validation loss decreased inf --> 5.04232: saving state...
Epoch 1/30 - Validation loss: 5.04232 (Exact: 0.00% | Partial: 0.00%)epoch time:03:18:07
█
Validation loss decreased 5.04232 --> 5.01229: saving state...
Epoch 2/30 - Validation loss: 5.01229 (Exact: 0.00% | Partial: 0.00%)epoch time:02:58:20
█
Validation loss decreased 5.01229 --> 4.99331: saving state...
Epoch 3/30 - Validation loss: 4.99331 (Exact: 0.00% | Partial: 0.00%)epoch time:02:58:53
█
Validation loss decreased 4.99331 --> 4.93169: saving state...
Epoch 4/30 - Validation loss: 4.93169 (Exact: 0.00% | Partial: 0.00%)epoch time:02:58:58
█
Epoch 5/30 - Validation loss: 5.08007 (Exact: 0.00% | Partial: 0.00%)epoch time:02:58:44
█
Epoch 6/30 - Validation loss: 5.26553 (Exact: 0.00% | Partial: 0.00%)epoch time:03:03:05
█
Epoch 7/30 - Validation loss: 5.25338 (Exact: 0.00% | Partial: 0.00%)epoch time:03:15:38
█
Epoch 8/30 - Validation loss: 5.3723 (Exact: 0.00% | Partial: 0.00%)epoch time:03:21:41
█
Epoch 9/30 - Validation loss: 5.42941 (Exact: 0.00% | Partial: 0.00%)epoch time:03:29:13
█
Epoch 10/30 - Validation loss: 5.19149 (Exact: 0.00% | Partial: 0.00%)epoch time:03:21:58
█
Epoch 11/30 - Validation loss: 5.22644 (Exact: 0.00% | Partial: 0.00%)epoch time:03:18:15
█
Validation loss decreased 4.93169 --> 4.46529: saving state...
Epoch 12/30 - Validation loss: 4.46529 (Exact: 0.00% | Partial: 0.00%)epoch time:03:21:03

**Reply by lfxuan** [reply]
This is mainly because the training data set is too complex. Using a simple data set to train a pre training model and lower learning rate training is helpful to train the model.
